lib/datasource/adoptium-java/index.ts

```diff
@@ -25,6 +25,31 @@ export class AdoptiumJavaDatasource extends Datasource {
 
   override readonly caching = true;
 
+  private async getPageReleases(
+    url: string,
+    page: number
+  ): Promise<Release[] | null> {
+    const pgUrl = `${url}&page=${page}`;
+    try {
+      const pgRes = await this.http.getJson<AdoptiumJavaResponse>(pgUrl);
+      return (
+        pgRes?.body?.versions?.map(({ semver }) => ({
+          version: semver,
+        })) ?? null
+      );
+    } catch (err) {
+      if (
+        page !== 0 &&
+        err instanceof HttpError &&
+        err.response?.statusCode === 404
+      ) {
+        return null;
```

Comment:
```suggestion
        // No more pages
        return null;
```
Add the lost comment